### PR TITLE
add equality to ore veins to hopefully avoid crashes

### PR DESCRIPTION
**Pull request by screret**
## What
this PR aims to stop crashes caused by just running /reload and the vein registry getting different values. I don't know if it works yet, someone will have to test.

## Implementation Details
Added an implementation of equals() and hashCode() to veins, indicators

## Outcome
no more crash (hopefully)

### CODE REVIEW

**Comment by a-a-GiTHuB-a-a** [comment]
How do I test this?

**Comment by Deepacat** [comment]
> How do I test this?

Go in singleplayer and use /reload then open interface a bunch? It may take having a custom ore vein registered or removal of a vein I'm unsure but these were the cases I experienced this in

File: src/main/java/com/gregtechceu/gtceu/api/data/worldgen/BiomeWeightModifier.java [hunk 1]
```diff
@@ -33,4 +34,19 @@ public BiomeWeightModifier(Supplier<HolderSet<Biome>> biomes, int addedWeight) {
     public Integer apply(Holder<Biome> biome) {
         return biomes.get().contains(biome) ? addedWeight : 0;
     }
+
+    @Override
+    public final boolean equals(Object object) {
+        if (this == object) return true;
+        if (!(object instanceof BiomeWeightModifier that)) return false;
+
+        return addedWeight == that.addedWeight && biomes.get().equals(that.biomes.get());
```
Comment (krossgg) [inline src/main/java/com/gregtechceu/gtceu/api/data/worldgen/BiomeWeightModifier.java:43]: Could the HolderSet supplied ever be null?

Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/api/data/worldgen/BiomeWeightModifier.java:43]: not in there, no.

**Review by krossgg — APPROVED** [review]
Good but wait for one of the 5 people who can reproduce to test before merge

**Pull request closed without merging**